src/Paramore.Brighter.MessagingGateway.AWSSQS/SqsInlineMessageCreator.cs

```diff
@@ -239,13 +240,19 @@ private HeaderResult<RoutingKey> ReadTopic()
     {
         if (_messageAttributes.TryGetValue(HeaderNames.Topic, out var topicArn))
         {
-            //we have an arn, and we want the topic
-            var s = topicArn.GetValueInString();
-            if (string.IsNullOrEmpty(s))
-                return new HeaderResult<RoutingKey>(RoutingKey.Empty, true);
-
-            var arnElements = s!.Split(':');
-            var topic = arnElements[(int)ARNAmazonSNS.TopicName];
+            var topic = topicArn.GetValueInString() ?? string.Empty;
```

Comment:
The earlier version reduces nesting by returning early, which is generally preferable